### PR TITLE
Make ivy-rich a layer variable

### PR DESCRIPTION
**Pull request by AlejandroCatalina**
Make ```ivy-rich``` a variable within the ```ivy``` layer to install it or not. Address #10101 .

It's been a while since my last PR, so if I forgot some style-related thing please tell me. Or whatever.

### CODE REVIEW

File: layers/+completion/ivy/config.el [hunk 1]
```diff
@@ -47,3 +47,6 @@ than this amount.")
         for desc = (nth 2 j)
         collect `(,key (lambda (x) (spacemacs//counsel-with-git-grep (quote ,func) x)) ,desc))
   "Default ivy actions to be used with git-grep output.")
+
+(defvar spacemacs--ivy-rich nil
```
Comment (nixmaniack) [inline layers/+completion/ivy/config.el]: This layer variable should be something like `ivy-enable-ivy-rich`. `spacemacs--*` are, I think, private/internal.

Comment (AlejandroCatalina) [reply layers/+completion/ivy/config.el]: Okay I doubted that, I'll update it. Thanks!

File: layers/+completion/ivy/config.el [hunk 2]
```diff
@@ -47,3 +47,6 @@ than this amount.")
...
+  "Whether include ivy-rich or not.")
```
Comment (nixmaniack) [inline layers/+completion/ivy/config.el]: Maybe "If non-nil, enable `ivy-rich'."

Comment (AlejandroCatalina) [reply layers/+completion/ivy/config.el]: Updated and thanks. I'll catch up on naming and doc conventions.

File: layers/+completion/ivy/packages.el [hunk 3]
```diff
@@ -219,15 +219,16 @@
...
-  (use-package ivy-rich
-    :defer t
-    :init
-    (progn
-      (setq ivy-rich-abbreviate-paths t
-            ivy-virtual-abbreviate 'full
-            ivy-rich-switch-buffer-align-virtual-buffer t)
-      (ivy-set-display-transformer 'ivy-switch-buffer
-                                   'ivy-rich-switch-buffer-transformer))))
+  (when ivy-enable-ivy-rich
```
Comment (nixmaniack) [inline layers/+completion/ivy/packages.el]: Sorry that I didn't use review system. Hopefully last comment!

You can just use something like this in the `(setq ivy-packages..` block to achieve this. Please test once.
```(ivy-rich :toggle ivy-enable-ivy-rich)```

Comment (AlejandroCatalina) [reply layers/+completion/ivy/packages.el]: Yep, that works.

**Comment by syl20bnr** [comment]
Thank you ! 👍 

I changed the name of the layer variable to `ivy-enable-advanced-buffer-information` as `ivy-rich` may support additional sources, it will allow users to select which sources they want to enhanced.
commit https://github.com/syl20bnr/spacemacs/commit/014ed731

Cherry-picked into develop branch, you can safely delete your branch.

**Pull request closed without merging**